### ISSUE TITLE
LLDB symlink issue in main_module_name

### ISSUE THREAD

**Issue by patryk4815** [pwndbg/pwndbg#2590]
```
(Pdb) pwndbg.dbg.selected_inferior().target.GetModuleAtIndex(0).GetFileSpec().fullpath
'/bin/sh'

(Pdb) pwndbg.dbg.selected_inferior().main_module_name()
'/bin/nix/store/gz1sk14pfgam3q6sgl5gfm1kdgpvvy6w-bash-interactive-5.2p26/bin/bash'
```

I have file `/bin/sh` and he is symlink to `/nix/store/gz1sk14pfgam3q6sgl5gfm1kdgpvvy6w-bash-interactive-5.2p26/bin/bash`
So `main_module_name` is resolving symlink wrong!

@mbrla0 Do you remember why we have here `{spec.dirname}`? Maybe there was reason/edge-case?
```py
        # We should resolve symbolic links.
        link = pwndbg.aglib.file.readlink(spec.fullpath)
        if len(link) == 0:
            return spec.fullpath

        return os.path.normpath(f"{spec.dirname}/{link}")
```

**Comment by mbrla0** [comment]
> Do you remember why we have here {spec.dirname}? Maybe there was reason/edge-case?

It's supposed to handle relative-pathed symlinks.

We could replace that with:
```py
return link if os.path.isabs(link) else os.path.normpath(f"{spec.dirname}/{link}")
```

Which should let us handle both.

**Comment by mbrla0** [comment]
Or maybe we could just use `os.path.realpath`, which can handle both..

**Comment by disconnect3d** [comment]
Fixed by #2591
